File: mobile/src/main/java/ro/antiprotv/sugar/MainPagerAdapter.java
package ro.antiprotv.sugar;

import android.content.Context;

import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

/**
 * A [FragmentPagerAdapter] that returns a fragment corresponding to
 * one of the sections/tabs/pages.
 */
public class MainPagerAdapter extends FragmentPagerAdapter {

    @StringRes
    private static final int[] TAB_TITLES = new int[]{R.string.title_alert_list, R.string.title_items, R.string.title_help};
    private final Context mContext;

    public MainPagerAdapter(Context context, FragmentManager fm) {
        super(fm);
        mContext = context;
    }

    @Override
    public Fragment getItem(int position) {
        Fragment f = AlertListFragment.newInstance();
        switch (position) {
            case 0:
                f = AlertListFragment.newInstance();
                break;
            case 1:
                f = ItemListFragment.newInstance();
                break;
        }
        return f;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return mContext.getResources().getString(TAB_TITLES[position]);
    }

    @Override
    public int getCount() {
        // total pages.
        return 3;
    }
}

File: mobile/src/main/java/ro/antiprotv/sugar/ui/MainPagerAdapter.kt
package ro.antiprotv.sugar.ui

import android.content.Context
import androidx.annotation.StringRes
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import org.koin.core.KoinComponent
import org.koin.core.get
import ro.antiprotv.sugar.R
import java.lang.IllegalArgumentException

/**
 * A [FragmentPagerAdapter] that returns a fragment corresponding to
 * one of the sections/tabs/pages.
 */
class MainPagerAdapter(private val mContext: Context, fm: FragmentManager) : KoinComponent,
        FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> get<AlertListFragment>()
            1 -> get<ItemListFragment>()
            // 2 -> TODO: help_fragment_here
            else -> throw IllegalArgumentException("Unsupported item position: $position")
        }
    }

    override fun getPageTitle(position: Int): CharSequence = mContext.resources.getString(TAB_TITLES[position])

    override fun getCount() = TAB_TITLES.size       // total pages.

    companion object {
        @StringRes
        private val TAB_TITLES = intArrayOf(R.string.title_alert_list, R.string.title_items/*, R.string.title_help*/)
    }

}